homeassistant/components/homekit_controller/event.py

```diff
@@ -0,0 +1,156 @@
+                self._handle_event,
+            )
+        )
+
+    @callback
+    def _handle_event(self):
+        self._trigger_event(INPUT_EVENT_VALUES[self._characteristic.value])
+        self.async_write_ha_state()
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Homekit event."""
+    hkid: str = config_entry.data["AccessoryPairingID"]
+    conn: HKDevice = hass.data[KNOWN_DEVICES][hkid]
+
+    @callback
+    def async_add_characteristic(char: Characteristic) -> bool:
+        service = char.service
+        entities = []
+
+        if service.type == ServicesTypes.DOORBELL:
+            entities.append(
+                HomeKitEventEntity(
+                    conn,
+                    service,
+                    EventEntityDescription(
+                        key=f"{service.accessory.aid}_{service.iid}",
+                        device_class=EventDeviceClass.DOORBELL,
+                        translation_key="doorbell",
+                        name="Doorbell",
+                    ),
+                )
+            )
+
+        elif service.type == ServicesTypes.SERVICE_LABEL:
+            switches = list(
+                service.accessory.services.filter(
+                    service_type=ServicesTypes.STATELESS_PROGRAMMABLE_SWITCH,
+                    child_service=service,
+                    order_by=[CharacteristicsTypes.SERVICE_LABEL_INDEX],
+                )
+            )
+
+            for idx, switch in enumerate(switches):
+                entities.append(
+                    HomeKitEventEntity(
+                        conn,
+                        switch,
+                        EventEntityDescription(
+                            key=f"{service.accessory.aid}_{service.iid}",
+                            device_class=EventDeviceClass.BUTTON,
+                            translation_key="button",
+                            name=f"Button {idx}",
+                        ),
+                    )
+                )
+
+        elif service.type == ServicesTypes.STATELESS_PROGRAMMABLE_SWITCH:
+            # A stateless switch that has a SERVICE_LABEL_INDEX is part of a group
+            # And is handled separately
+            if not service.has(CharacteristicsTypes.SERVICE_LABEL_INDEX):
+                entities.append(
+                    HomeKitEventEntity(
+                        conn,
+                        service,
+                        EventEntityDescription(
+                            key=f"{service.accessory.aid}_{service.iid}",
+                            device_class=EventDeviceClass.BUTTON,
+                            translation_key="button",
+                            name="Button",
+                        ),
+                    )
+                )
+
+        if entities:
+            async_add_entities(entities)
+            return True
+
+        return False
```

Comment:
```suggestion
        if entities:
            async_add_entities(entities)

```

typed to `None` but returning bool

Reply:
The inner function is typed to bool, the outer function is typed to none. It only marks a characteristic as processed if the callback returns true.